### PR TITLE
Update less serializations KEP based on final implementation

### PR DESCRIPTION
**Pull request by wojtek-t**
/assign @liggitt

### CODE REVIEW

File: keps/sig-api-machinery/20190329-less-object-serializations.md [hunk 1]
```diff
@@ -120,91 +120,106 @@ read/watched by different components.
...
+// SelfLink (which is set lately in the path) are ignored.
+type cachingObject struct {
+	lock sync.RWMutex
+
+	// Object for which serializations are cached.
+	object metaRuntimeInterface
+
+	// serializations is a cache containing object`s serializations.
+	// The value stored in atomic.Value is of type serializationsCache.
+	// The atomic.Value type is used to allow fast-path.
```
Comment (lavalamp) [inline keps/sig-api-machinery/20190329-less-object-serializations.md]: What type is stored in the atomic.Value?

**Comment by lavalamp** [comment]
/approve

I won't hold this up for my question.

I know it's for a good cause but this really did not improve the understandability of our encoding stack, I'm kind of sad about that, it was already very difficult to understand :(

File: keps/sig-api-machinery/20190329-less-object-serializations.md (hunk 1 above)
Comment (wojtek-t) [reply keps/sig-api-machinery/20190329-less-object-serializations.md]: The line above is saying "The value stored in atomic.Value is of type serializationsCache."

Apparently I forgot to copy the serializationsCache type from the PR:
```
// serializationsCache is a type for caching serialization results.
type serializationsCache map[runtime.Identifier]*serializationResult
```

**Pull request merged**